server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -324,7 +360,7 @@ public SourceFieldMapper build() {
             }
 
             SourceFieldMapper sourceFieldMapper = new SourceFieldMapper(
-                sourceMode,
+                INDEX_MAPPER_SOURCE_MODE_SETTING.exists(settings) ? INDEX_MAPPER_SOURCE_MODE_SETTING.get(settings) : mode.get(),
```

Comment:
This is not really required. Will remove.

Reply:
I think need this logic here? Otherwise we never use source mode defined in the _source field mapper?

Reply:
This is already done above

```
final Mode sourceMode = INDEX_MAPPER_SOURCE_MODE_SETTING.exists(settings)
                ? INDEX_MAPPER_SOURCE_MODE_SETTING.get(settings)
                : mode.get();
                ```

Reply:
Ok, I missed that. 👍